src/__experimental__/components/date/date-picker.component.js

```diff
@@ -58,15 +58,18 @@ const DatePicker = ({
     fixedWeeks: true,
     initialMonth: selectedDate || undefined,
     inline: true,
-    locale: I18n.locale,
+    locale: i18n.language,
     localeUtils: LocaleUtils,
     navbarElement: <Navbar />,
     onDayClick: handleDayClick,
     selectedDays: selectedDate || undefined,
     weekdayElement: (weekdayElementProps) => {
       const { className, weekday, localeUtils } = weekdayElementProps;
-      const weekdayLong = localeUtils.formatWeekdayLong(weekday, I18n.locale);
-      const weekdayShort = localeUtils.formatWeekdayShort(weekday, I18n.locale);
+      const weekdayLong = localeUtils.formatWeekdayLong(weekday, i18n.language);
+      const weekdayShort = localeUtils.formatWeekdayShort(
+        weekday,
+        i18n.language
```

Comment:
We'll have to ensure the context provider also provides language, as well as `t`.